File: le-transform/src/main/java/com/latticeengines/transform/v2_0_25/functions/StdVisidbDsStateIsInNewEngland.java
package com.latticeengines.transform.v2_0_25.functions;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Map;

import org.apache.commons.lang3.StringUtils;

import com.latticeengines.domain.exposed.metadata.ApprovedUsage;
import com.latticeengines.domain.exposed.metadata.Attribute;
import com.latticeengines.domain.exposed.metadata.Category;
import com.latticeengines.domain.exposed.metadata.FundamentalType;
import com.latticeengines.domain.exposed.metadata.StatisticalType;
import com.latticeengines.domain.exposed.metadata.Tag;
import com.latticeengines.transform.exposed.RealTimeTransform;

public class StdVisidbDsStateIsInNewEngland implements RealTimeTransform {

    private static final long serialVersionUID = -6044120647859842967L;
    static HashSet<String> valueMap = new HashSet<String>(Arrays.asList("ME", "NH", "MA", "VT", "RI", "CT"));

    public StdVisidbDsStateIsInNewEngland() {
        
    }
    
    public StdVisidbDsStateIsInNewEngland(String modelPath) {
    }

    public Object transform(Map<String, Object> arguments, Map<String, Object> record) {
        String column = (String) arguments.get("column");
        Object n = record.get(column);

        if (n == null)
            return false;

        String s = n.toString().trim().toUpperCase();

        return getDSStateIsInNewEngland(s);
    }

    public static Boolean getDSStateIsInNewEngland(String state) {
        if (StringUtils.isEmpty(state))
            return false;

        return valueMap.contains(state);
    }

    @Override
    public Attribute getMetadata() {
        Attribute attribute = new Attribute();
        attribute.setApprovedUsage(ApprovedUsage.MODEL_ALLINSIGHTS);
        attribute.setCategory(Category.LEAD_INFORMATION);
        attribute.setFundamentalType(FundamentalType.BOOLEAN);
        attribute.setStatisticalType(StatisticalType.NOMINAL);
        attribute.setDescription("Region: New England");
        attribute.setDisplayName("Region: New England");
        attribute.setTags(Tag.INTERNAL_TRANSFORM);
        return attribute;
    }
}

File: le-serviceflows/core/src/main/resources/com/latticeengines/serviceflows/core/transforms/std_visidb_ds_state_isInNewEngland.py
#This function looks for words and returns TRUE if there is a word map, FALSE otherwise

def metadata():
    return {  'ApprovedUsage'   : 'ModelAndAllInsights'
            , 'Category'        : 'Lead Information'
            , 'FundamentalType' : 'boolean'
            , 'Description'     : 'Region: New England'
            , 'DisplayName'     : 'Region: New England'            
            , 'StatisticalType' : 'nominal'
            , 'Tags'            : 'Internal' }

#simple test for membership
def std_visidb_ds_state_isInNewEngland(x):
    return x in set(['ME', 'NH', 'MA', 'VT', 'RI', 'CT'])

#function to call
def transform(args, record):
    column = args["column"]
    value = record[column]
    return std_visidb_ds_state_isInNewEngland(value)
